### PR TITLE
Drawer gap issue Fix

### PR DESCRIPTION
**Pull request by Anubhvv**
### Problem 
navigation bar and left/ tight drawer were causing issues
### Root cause 
In horizontal drawers, we set shadow elevation = 0
### Validations

(how the change was tested, including both manual and automated tests)

### Screenshots

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot_2023-09-04-14-21-57-53_ad3b258aa0cd3ec091b47fe7b71fec5b | Screenshot_2023-09-04-16-01-25-09_138c6ebfedc8e8a8cd1938476bab217a|
|Screenshot_2023-09-04-16-01-07-60_ad3b258aa0cd3ec091b47fe7b71fec5b|Screenshot_2023-09-04-16-01-41-87_138c6ebfedc8e8a8cd1938476bab217a|

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] Automated Tests
- [ ] Documentation and demo app examples
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt [hunk 1]
```diff
@@ -890,7 +891,7 @@ fun Drawer(
                     modifier = modifier,
                     drawerState = drawerState,
                     drawerShape = drawerShape,
-                    drawerElevation = drawerElevation,
+                    drawerElevation = horizontalDrawerElevation,
```
Comment (ayush547) [inline fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:894]: If HorizontalDrawer is intended to be used with just 0 dp elevation, maybe you can remove the drawerElevation parameter itself?

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt [hunk 2]
```diff
@@ -849,6 +849,7 @@ fun Drawer(
                     else -> RoundedCornerShape(tokens.borderRadius(drawerInfo))
                 }
             val drawerElevation: Dp = tokens.elevation(drawerInfo)
+            val horizontalDrawerElevation: Dp = 0.dp
```
Comment (ayush547) [inline fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:852]: Add a small comment explaining the reason for 0 elevation

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ToolTipActivity.kt [hunk 3]
```diff
@@ -133,30 +141,47 @@ fun CreateToolTipActivityUI(context: Context) {
...
+                            keyboardOptions = KeyboardOptions(keyboardType=KeyboardType.Text)
+                        )
+                    }
                 })
             ListItem.Header(title = context.getString(R.string.tooltip_text),
                 trailingAccessoryContent = {
-                    BasicTextField(
-                        value = toolTipText.value,
-                        onValueChange = { toolTipText.value = it })
+
```
Comment (ayush547) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ToolTipActivity.kt:177]: nit: extra line

**Comment by ayush547** [comment]
Can you update pr desc to something more verbose instead of -
`navigation bar and left/ tight drawer were causing issues`

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt (hunk 2 above)
Comment (mishramayank1) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:852]: Set the default value for elevation in token to 0.dp. Ask designer to update the same in Figma spec sheet.

Comment (mishramayank1) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:852]: Remove the horizontalDrawerElevation, the issue also exists in BottomDrawer as well.

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ToolTipActivity.kt
Comment (mishramayank1) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ToolTipActivity.kt:1]: remove this file as it is already a part of another PR

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt (hunk 2 above)
Comment (Anubhvv) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt]: Hi, due to version upgrade, I have created a new PR for this issue: https://github.com/microsoft/fluentui-android/pull/501
I have tried to incorporate all your suggestions.

**Pull request closed without merging**